### DISCUSSION TITLE
Single Use Keys

### DISCUSSION THREAD

**Discussion by WillXUMM** [XRPLF/XRPL-Standards#93] [XLS Ideas (pre standard proposal)]
**Work in progress**

Generating single use keys XRPL via derivation paths

Entropy --> Secret numbers / Family Seed / Mnemonic -> private key --> public key --> address

vs. 

Entropy --> Secret numbers / Family Seed / Mnemonic -> private key --> derivation path
                                                                                                                                   -> public key --> address 1
                                                                                                                                   -> public key --> address 2
                                                                                                                                   -> public key --> address 3
                                                                                                                                   -> and so on

A Hierarchical Deterministic (HD) wallet uses a private key to derive public keys and (via hash function) public addresses.

**Comment by elmurci** [comment]
Quick question for my own understanding if I may :-)

How would this be different to what is already supported?

`xrpl.Wallet.fromMnemonic(<mnemonic>, { derivationPath: <derivationPath> });`
